### ISSUE TITLE
Autostart of Everything when it's not launched

### ISSUE THREAD

**Issue by RadRussianRus** [stnkl/EverythingToolbar#75]
For me Everything eats ~250 MB RAM, so I launch only when it's needed instead of putting it to autostart. Since you can choose Everything.exe for "Show in Everything", could be same launched in tray only when opened?

For example:
1. Windows booting up, Everything not launched
2. Open EverythingToolbar or start typing
3. After that, Everyting is started on background
4. Results are shown when Everything is loaded

Or there is another way:
1. Windows booting up, Everything not launched
2. Open EverythingToolbar or start typing
3. Search window shows "Everything not started. Run", where run is a button or link that launches Everything in background.
4. When "Run" is clicked, Everyting is started on background
5. Results are shown when Everything is loaded

#### Some options that may be useful

- `-startup`: run Everything in background.
- `-first-instance`: don't run Everything if it's already started.

**Comment by perplexityjeff** [comment]
This is a good idea to implement when the form opens to search. I have the same issue as well and it shouldn't be to hard to code. 

1. Detect if Everything is running or not. This should be easy as the process name is 'Everything'
2. Start Everything with process argument '-startup'

This all could happen without the user noticing anything.

**Comment by Horst12** [comment]
> 
> 
> I love the bar in the state that it is right now so thank you for all the work so far.
> 
> This is a good idea to implement when the form opens to search. I have the same issue as well and it shouldn't be to hard to code.
> 
>     1. Detect if Everything is running or not. This should be easy as the process name is 'Everything'
> 
>     2. Start Everything with process argument '-startup'
> 
> 
> This all could happen without the user noticing anything. The hard thing is that the installation location is not always the same which makes it interesting how to solve this but I don't know all the locations by heart.

On start-up Everything will generate or update the index which may take a long time.
The real benefit of Everything comes only by always running in the background.

**Comment by perplexityjeff** [comment]
> > I love the bar in the state that it is right now so thank you for all the work so far.
> > This is a good idea to implement when the form opens to search. I have the same issue as well and it shouldn't be to hard to code.
> > ```
> > 1. Detect if Everything is running or not. This should be easy as the process name is 'Everything'
> > 
> > 2. Start Everything with process argument '-startup'
> > ```
> > 
> > 
> > This all could happen without the user noticing anything. The hard thing is that the installation location is not always the same which makes it interesting how to solve this but I don't know all the locations by heart.
> 
> On start-up Everything will generate or update the index which may take a long time.
> The real benefit of Everything comes only by always running in the background.

That is true however I have Everything as a service installed indexing and EverythingToolbar cannot communicate using that without the UI being open in the background.

**Comment by Horst12** [comment]
> 
> 
> > > I love the bar in the state that it is right now so thank you for all the work so far.
> > > This is a good idea to implement when the form opens to search. I have the same issue as well and it shouldn't be to hard to code.
> > > ```
> > > 1. Detect if Everything is running or not. This should be easy as the process name is 'Everything'
> > > 
> > > 2. Start Everything with process argument '-startup'
> > > ```
> > > 
> > > 
> > > This all could happen without the user noticing anything. The hard thing is that the installation location is not always the same which makes it interesting how to solve this but I don't know all the locations by heart.
> > 
> > 
> > On start-up Everything will generate or update the index which may take a long time.
> > The real benefit of Everything comes only by always running in the background.
> 
> That is true however I have Everything as a service installed indexing and EverythingToolbar cannot communicate using that without the UI being open in the background in my experience.

The Everything service only handles the UAC and does not index at all !
Of course the Toolbar needs Everything running in the background
and if it would only start on demand by the Toolbar a long delay may occur.

**Comment by stnkl** [comment]
I think the way Everything/EverythingToolbar is used should be up to the user so I support the suggestion to automatically launch Everything if it is not already running. A startup delay can be handled by displaying a waiting animation within EverythingToolbar.

**Comment by perplexityjeff** [comment]
> > > > I love the bar in the state that it is right now so thank you for all the work so far.
> > > > This is a good idea to implement when the form opens to search. I have the same issue as well and it shouldn't be to hard to code.
> > > > ```
> > > > 1. Detect if Everything is running or not. This should be easy as the process name is 'Everything'
> > > > 
> > > > 2. Start Everything with process argument '-startup'
> > > > ```
> > > > 
> > > > 
> > > > This all could happen without the user noticing anything. The hard thing is that the installation location is not always the same which makes it interesting how to solve this but I don't know all the locations by heart.
> > > 
> > > 
> > > On start-up Everything will generate or update the index which may take a long time.
> > > The real benefit of Everything comes only by always running in the background.
> > 
> > 
> > That is true however I have Everything as a service installed indexing and EverythingToolbar cannot communicate using that without the UI being open in the background in my experience.
> 
> The Everything service only handles the UAC and does not index at all !
> Of course the Toolbar needs Everything running in the background
> and if it would only start on demand by the Toolbar a long delay may occur.

If I take a look into the documentation this is not the case? At least I get the impression that it does help the indexing. 
https://www.voidtools.com/support/everything/everything_service/

**Comment by Horst12** [comment]
> 
> 
> @Horst12 If I take a look into the documentation this is not the case? At least I get the impression that it does help the indexing.
> https://www.voidtools.com/support/everything/everything_service/
> 
> @stnkl Thank you, I agree.
Belive me, the service doesn't index (I use Everything since its very first versions years ago).
The service only helps by avoiding the UAC prompt which would be necessary
if Everything isn't run as Administrator, which is not suggested.

**Comment by yfdyh000** [comment]
In addition to auto-start, I might prefer to provide a not-for-ready tip as well as a start button.
